skpro/regression/gradient_boosting/_ngboost.py

```diff
@@ -0,0 +1,353 @@
+"""Adapters to ngboost regressors with probabilistic components."""
+# copyright: skpro developers, BSD-3-Clause License (see LICENSE file)
+
+__author__ = ["ShreeshaM07"]
+
+import numpy as np
+from sklearn.utils import check_random_state
+
+from skpro.regression.base import BaseProbaRegressor
+
+
+class NGBoostRegressor(BaseProbaRegressor):
+    """
+    Constructor for all NGBoost models.
```

Comment:
Here you probably want to describe the regressor instead, i.e., it is an interface to `NGBRegressor`.